modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java

```diff
@@ -0,0 +1,384 @@
+
+        private SimpleIgniteNode(int i) {
+            nodeName = nodeName(i);
+            nodeDir = workDir.resolve(nodeName);
+
+            assertThat(networkConfiguration.port().update(port(i)), willCompleteSuccessfully());
+
+            var clusterLocalConfiguration = new ClusterLocalConfiguration(nodeName, defaultSerializationRegistry());
+
+            var nettyBootstrapFactory = new NettyBootstrapFactory(networkConfiguration, clusterLocalConfiguration.getName());
+
+            nettyBootstrapFactory.start();
+            cleanup.add(nettyBootstrapFactory::stop);
+
+            clusterSvc = new TestScaleCubeClusterServiceFactory().createClusterService(
+                    clusterLocalConfiguration,
+                    networkConfiguration,
+                    nettyBootstrapFactory
+            );
+
+            clusterSvc.start();
+            cleanup.add(clusterSvc::stop);
+
+            raftMgr = new Loza(clusterSvc, raftConfiguration, nodeDir, hybridClock);
+
+            raftMgr.start();
+            cleanup.add(raftMgr::stop);
+
+            cleanup.add(this::stopService);
+        }
+
+        void startService() {
+            // Refresh listener, to forget the old "last value".
+            raftGroupListener = new TestRaftGroupListener();
+
+            try {
+                serviceFuture = raftMgr.startRaftGroupNode(
+                        new RaftNodeId(GROUP_ID, new Peer(nodeName)),
+                        raftGroupConfiguration,
+                        raftGroupListener,
+                        RaftGroupEventsListener.noopLsnr,
+                        RaftGroupOptions.defaults().setLogStorageFactory(logStorageFactory)
+                );
+            } catch (NodeStoppingException e) {
+                fail(e.getMessage());
+            }
+        }
+
+        RaftGroupService getService() {
+            assertNotNull(serviceFuture);
+            assertThat(serviceFuture, willCompleteSuccessfully());
+
+            return serviceFuture.join();
+        }
+
+        void stopService() {
+            if (serviceFuture != null) {
+                serviceFuture = null;
+
+                try {
+                    raftMgr.stopRaftNode(new RaftNodeId(GROUP_ID, new Peer(nodeName)));
+                } catch (NodeStoppingException e) {
+                    fail(e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Tests a situation when one of the nodes in RAFT group lost its last update upon restart.
+     */
+    @Test
+    void testRestartSingleNode() throws Exception {
+        SimpleIgniteNode node0 = nodes.get(0);
+        SimpleIgniteNode node1 = nodes.get(1);
+        SimpleIgniteNode node2 = nodes.get(2);
+
+        assertThat(node1.getService().run(testWriteCommand("1")), willCompleteSuccessfully());
+        waitForValueReplication(nodes, "1");
+
+        node1.stopService();
+        node1.logStorage.truncateSuffix(1);
+        node1.startService();
```

Comment:
Shouldn't you wait for the node to finish starting up?

Reply:
No, following waiting of the log replication is enough